cmake/extensions.cmake

```diff
@@ -2261,3 +2263,368 @@ function(target_byproducts)
+
+########################################################
+# 4. Zephyr devicetree function
+########################################################
+# 4.1. dt_*
+#
+# The following methods are for retrieving devicetree information in CMake.
+#
+# Note: In CMake we refer to the nodes using the node's path, therefore there
+# is no dt_path(...) function for obtaining a node identifier.
+
+# Usage:
+#   dt_nodelabel(<var> LABEL <label>)
+#
+# function for retrieving the node path for the node having nodelabel
+# <label>.
+#
+# the node path will be returned in the <var> parameter.
+#
+# <var>          : Return variable where the node path will be stored
+# LABEL <label>  : Node label
+function(dt_nodelabel var)
+  set(req_single_args "LABEL")
+  cmake_parse_arguments(DT_LABEL "" "${req_single_args}" "" ${ARGN})
+
+  if(${ARGV0} IN_LIST req_single_args)
+    message(FATAL_ERROR "dt_nodelabel(${ARGV0} ...) missing return parameter.")
+  endif()
+
+  foreach(arg ${req_single_args})
+    if(NOT DEFINED DT_LABEL_${arg})
+      message(FATAL_ERROR "dt_nodelabel(${ARGV0} ...) "
+                          "missing required argument: ${arg}"
+      )
+    endif()
+  endforeach()
+
+  get_target_property(${var} devicetree_target "DT_LABEL|${DT_LABEL_LABEL}")
+  if(${${var}} STREQUAL ${var}-NOTFOUND)
+    set(${var})
+  endif()
+
+  set(${var} ${${var}} PARENT_SCOPE)
+endfunction()
+
+# Usage:
+#   dt_node_exists(<var> NODE <node>)
+#
+# Does a node path refer to a node?
+#
+# Tests whether a node path refers to a node which exists, i.e. is defined
+# in the devicetree.
+#
+# the result of the check will be returned in the <var> parameter.
+#
+# <var>       : Return variable where the check result will be returned
+# NODE <node> : Node path
+function(dt_node_exists var)
+  set(req_single_args "NODE")
+  cmake_parse_arguments(DT_NODE "" "${req_single_args}" "" ${ARGN})
+
+  if(${ARGV0} IN_LIST req_single_args)
+    message(FATAL_ERROR "dt_node_existsl(${ARGV0} ...) missing return parameter.")
+  endif()
+
+  foreach(arg ${req_single_args})
+    if(NOT DEFINED DT_NODE_${arg})
+      message(FATAL_ERROR "dt_node_exists(${ARGV0} ...) "
+                          "missing required argument: ${arg}"
+      )
+    endif()
+  endforeach()
+
+  get_target_property(${var} devicetree_target "DT_NODE|${DT_NODE_NODE}")
+
+  if(${var})
+    set(${var} ${${var}} PARENT_SCOPE)
+  else()
+    set(${var} FALSE PARENT_SCOPE)
+  endif()
+endfunction()
+
+# Usage:
+#   dt_node_has_status(<var> NODE <node> STATUS <status>)
+#
+# Tests whether a node path refers to a node which:
+# - exists in the devicetree, and
+# - has a status property matching the second argument
+#   (except that either a missing status or an “ok” status in the devicetree
+#    is treated as if it were “okay” instead)
+#
+# the result of the check will be returned in the <var> parameter.
+#
+# <var>           : Return variable where the check result will be returned
+# NODE <node>     : Node path
```

Comment:
I think this should be `PATH <path>` for now; we can extend it to `ALIAS`, etc. later. Similar comments to other functions.

Reply:
Changed all `NODE <path>` to `PATH <path>`.